### ISSUE TITLE
Draw image over rectangle

### ISSUE THREAD

**Issue by curtiskoelling** [signintech/gopdf#63]
I am trying to insert an image on top of a rectangle without much success.

```
pdf.SetFillColor(37, 45, 66)
pdf.RectFromLowerLeftWithStyle(X, Y+H, W, H, "F")

bytesOfImg := GetImageBytesFromOutThere()
imgH, _ := gopdf.ImageHolderByBytes(bytesOfImg)
imgRect := gopdf.Rect{
	W: 200,
	H: 300,
}
pdf.ImageByHolder(imgH, x, y, &imgRect)
```

No matter what I try the picture is drawn under the rectangle.

![layer](https://user-images.githubusercontent.com/5056768/28288389-41dcd318-6afc-11e7-8845-f96f21311089.png)

**Comment by oneplus1000** [comment]
I will look into it.

**Comment by oneplus1000** [comment]
fixed 722aebb74e6ff6fbb6f5845637b4190ad069ed0c
Thank you for your issue report. :smile:

**Comment by curtiskoelling** [comment]
Awesome, thanks for the speedy fix. Works great!

**Comment by curtiskoelling** [comment]
Bug was introduced by this fix. 
Same image repeats over and over again.
The following change fixes the issue:
```
diff --git a/content_obj.go b/content_obj.go
index 746e68f..bee611e 100644
--- a/content_obj.go
+++ b/content_obj.go
@@ -282,6 +282,7 @@ func (c *ContentObj) AppendStreamImage(index int, x float64, y float64, rect *Re
 	cache.x = x
 	cache.y = y
 	cache.rect = *rect
+	cache.index = index
 	c.listCache.append(&cache)
 	//c.stream.WriteString(fmt.Sprintf("q %0.2f 0 0 %0.2f %0.2f %0.2f cm /I%d Do Q\n", rect.W, rect.H, x, h-(y+rect.H), index+1))
 }
```

**Comment by oneplus1000** [comment]
fixed 35478c91826f8a133d1779ed8e4c3c0b36d6b388
Thank you.
